ci-operator/step-registry/ibmcloud/provision/dedicatedhost/ibmcloud-provision-dedicatedhost-commands.sh

```diff
@@ -0,0 +1,217 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+#Since sometimes fail to create the dedicated host in some zone, so just continue test when create dedicated host succeed, and create cluster just on one zone. 
+#Known Issues: 
+# OCPBUGS-5906 [IPI-IBMCloud] the worker nodes are “Provisioning” when using the dedicated host in the Default group
+# OCPBUGS-18925 [IPI-IBMCloud] ] fail to use vx2d-host-176x2464 profile with dedicated host in install-config
+
+function run_command() {
+    local CMD="$1"
+    echo "Running Command: ${CMD}"
+    eval "${CMD}"
+}
+
+# IBM Cloud CLI login
+function ibmcloud_login {
+    export IBMCLOUD_CLI=ibmcloud
+    export IBMCLOUD_HOME=/output
+    region="${LEASED_RESOURCE}"
+    export region
+    echo "Try to login..."
+    "${IBMCLOUD_CLI}" login -r ${region} --apikey @"${CLUSTER_PROFILE_DIR}/ibmcloud-api-key"
+}
+
+
+function create_resource_group() {
+    local rg="$1"
+    echo "create resource group ... ${rg}"
+    "${IBMCLOUD_CLI}" resource group-create ${rg} || return 1
+    echo "${rg}" > "${SHARED_DIR}/ibmcloud_resource_group"
+    "${IBMCLOUD_CLI}" target -g ${rg} || return 1
+}
+
+#random select the group which create the dedicatd host
+function randomSelectRG {
+    dhGroups=("Default" "${resource_group}")
+    selected_idx=$((RANDOM % 2))
+    rg=${dhGroups[selected_idx]}
+    echo $rg
+}
+
+function randomGetDHProfile {
+    local all_profiles profiles selected_id proTotal profile tmp family i
+    all_profiles=$(${IBMCLOUD_CLI} is dhps -q | awk '(NR>1) {print $1}')
+    mapfile -t profiles <<< "${all_profiles}"
+    proTotal=${#profiles[@]}
+    i=0
+    while [ $i -le ${proTotal} ]
+    do
+        i=$(($i+1))
+        selected_id=$((RANDOM % ${proTotal}))
+        profile=${profiles[$selected_id]}
+        family=$(${IBMCLOUD_CLI} is dedicated-host-profile $profile --output JSON | jq -r '.family')
+        if [[ ${family} != 'very-high-memory' ]]; then
+            echo ${profile}
+            return 0
+        fi
+    done
+    echo "ERROR: fail to get the DH profile"
+    return 1
+}
+
+function createDH() {
+    local rg=$1 dProfile=$2 zone=$3 dhName=$4
+    local tmp family class dhg status
+    if [[ ${rg} = "Default" ]]; then
+        run_command "${IBMCLOUD_CLI} target --unset-resource-group"
+    else
+        run_command "${IBMCLOUD_CLI} target -g $rg"
```

Comment:
if we always need to consider the current resource group, I would suggest not globally set target resource group, but add the resource group in the imbcloud command line.

Reply:
just create dhg (dedicated host group) need take care of whether created in "Default" group

Reply:
I am feeling there is design issue, from my understanding, from the very beginning, the target resource group should be already decided, in the following steps, just get the target resource group, and set target, but not judge the target resource group is "default" or not, then run some different commands.